lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -295,251 +320,57 @@ def prepare(self):
-        ipathpath = None
-        for ipp in ipathpaths:
-            if os.path.exists(ipp):
-                ipathpath = ipp
-                break
-
-        if ipathpath:
-            # how many contexts?
-            contxts = 0
-            sharedcontexts = True
-            sysib = "/sys/class/infiniband"
-            hcas = 0
-            if os.path.isdir(sysib):
-                for qibdir in os.listdir(sysib):
-                    fn = os.path.join(sysib, qibdir, 'nctxts')
-                    if qibdir.startswith('qib') and os.path.exists(fn):
-                        contxts += int(open(fn).read())
-                        hcas += 1
-            if contxts <= self.ppn:
-                # enough HW contexts not to share (assuming this is the only job on the node)
-                sharedcontexts = False
-
-            self.log.debug("Found %s HCAs with %s contexts for %s ppn: detected shared context %s" %
-                           (hcas, contxts, self.ppn, sharedcontexts))
-            self.mpiexec_global_options['PSM_SHAREDCONTEXTS'] = '%d' % sharedcontexts
-            if self.options.debuglvl > 0:
-                self.mpiexec_global_options['PSM_TRACEMASK'] = '0x101'
-
-            # not processed, so None by default. if so, PSM takes over
-            # IPATH_NO_CPUAFFINITY means: Prevent PSM from setting affinity?
-            if self.options.pinmpi:
-                self.mpiexec_global_options['IPATH_NO_CPUAFFINITY'] = '1'
-            else:
-                # Don't set the variable. The existince, not the value is checked.
-                # (Although traceback reports it properly, the actual process of setting it just checks the existence)
-                # self.mpiexec_global_options['IPATH_NO_CPUAFFINITY'] = '0'
-                self.options.pinmpi = False
-
-            self.log.debug("qlogic_ipath: ipath found %s" % ipathpath)
-            self.options.qlogic_ipath = True
-        else:
-            if self.options.qlogic_ipath:
-                self.log.debug("qlogic_ipath: forced ipath, but ipath path not found from paths %s" % ipathpaths)
-            else:
-                self.log.debug("qlogic_ipath: ipath path not found from paths %s" % ipathpaths)
-                self.options.qlogic_ipath = False
-
-    def scalemp_vsmp(self):
-        """See if the node is using ScaleMP vSMP to set various parameters
-            Detect vSMP presence + set additional default variables
-            - vsmpctl --features works
-            -- newer releases it is vsmpctl --status
-        """
-        setattr(self.options, 'scalemp_vsmp', None)
+        setattr(self.options, 'ompthreads', threads)
 
-        vsmpctl = "vsmpctl --status"
-        ec, out = run_simple_noworries(vsmpctl)
-        if ec > 0:
-            self.log.debug("scalemp_vsmp: vSMP not found (cmd %s ec %s output %s)" % (vsmpctl, ec, out))
-            return
+    def set_netmask(self):
+        """set self.netmask to a list containing (ip address/netmask)
```

Comment:
start comment at new line, start sentence with a capital. can you check at least the whole PR?